File: app/src/androidTest/java/com/github/fo2rist/mclaren/tests/CircuitsPageTest.java
package com.github.fo2rist.mclaren.tests;

import android.support.test.runner.AndroidJUnit4;

import com.github.fo2rist.mclaren.R;
import com.github.fo2rist.mclaren.pages.CircuitsPage;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.action.ViewActions.scrollTo;
import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;

@RunWith(AndroidJUnit4.class)
public class CircuitsPageTest extends BaseMainActivityTest {
    private CircuitsPage circuitsPage = new CircuitsPage();

    @Override
    @Before
    public void setUp() {
        super.setUp();
        mainPage.navigateToMenuItem(R.id.nav_circuits);
    }

    @Test
    public void testTopItemPresent() throws Exception {
        circuitsPage.onCircuitsList()
                .perform(actionOnItemAtPosition(0, click()));

    }

    @Test
    public void testBottomItemPresent() throws Exception {
        circuitsPage.onCircuitsList()
                .perform(actionOnItemAtPosition(20, scrollTo()))
                .perform(actionOnItemAtPosition(20, click()));
    }

}

File: app/src/androidTest/java/com/github/fo2rist/mclaren/tests/CircuitsPageTest.kt
package com.github.fo2rist.mclaren.tests

import androidx.test.espresso.intent.Intents.intended
import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.github.fo2rist.mclaren.R
import com.github.fo2rist.mclaren.pages.CircuitsPage
import com.github.fo2rist.mclaren.ui.circuitsscreen.CircuitDetailsActivity
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class CircuitsPageTest : BaseMainActivityTest() {

    private val circuitsPage = CircuitsPage()

    @Before
    fun setUp() {
        mainPage.navigateToMenuItem(R.id.nav_season_calendar)
    }

    @Test
    fun testTopItemPresent_andLaunchesDetails() {
        interceptIntents()

        circuitsPage {
            onCircuitsList.childAt<CircuitsPage.CircuitItem>(0) { click() }
        }

        intended(hasComponent(CircuitDetailsActivity::class.java.name))
    }

    @Test
    fun testBottomItemPresent_andLaunchesDetails() {
        interceptIntents()

        circuitsPage {
            onCircuitsList.childAt<CircuitsPage.CircuitItem>(15) { click() } // usually 20+, but not in 2020
        }

        intended(hasComponent(CircuitDetailsActivity::class.java.name))
    }
}
